camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelSimpleAnnotator.java

```diff
@@ -69,9 +69,14 @@ void validateText(@NotNull PsiElement element, @NotNull AnnotationHolder holder,
                     if (!result.isSuccess()) {
                         String error = result.getShortError();
                         TextRange range = element.getTextRange();
+                        int startIdx = result.getIndex() == 0 ? text.indexOf("$") : result.getIndex();
```

Comment:
Can you add some code comments to this code, its hard to follow what goes on, with the various things. Yes writing code with indexOf and substrings tend to be a bit unclear.